docs/HOL_Deploy_Private_Docker_Registry/README.md

```diff
@@ -0,0 +1,447 @@
+
+![](<media/scansecurity.png>)
+
+* **Action**: Select **Run a Docker command**.
+* **Command**: Enter the following command line:  
+
+    ```
+    run --net host --pid host --cap-add audit_control -v /var/lib:/var/lib -v /var/run/docker.sock:/var/run/docker.sock -v /usr/lib/systemd:/usr/lib/systemd -v /etc:/etc --label docker_bench_security docker/docker-bench-security
+    ```
+
+**Step 15.** Add a build step to push the **Clients** image to your private Docker registry. Configure the Docker task (e.g., Push Clients Image to Private Docker Registry) as follows:
+
+![](<media/pushclientsimage.png>)
+
+* **Docker Registry Connection**: Select the Docker registry endpoint created earlier.
+* **Action**: Select **Push an image**.
+* **Image Name**: Enter the image name (e.g., **clients:$(Build.BuildId)**) you wish to push.
+* **Qualify Image Name**: Tick this checkbox. Qualify the image name with the Docker registry connection's hostname.
+
+**Step 16.** Add a build step to remove the running container after testing. Configure the Docker task (e.g., Remove Clients Container after Testing) as follows:
+
+![](<media/removetestingclientscontainer.png>)
+
+* **Action**: Select **Run a Docker command**.
+* **Command**: Enter the following command line, and replace the container-name with your container name (e.g., clients):  
+
+    ```
+    rm -f <container-name>
+    ```
+
+**Step 17.** Add a build step to remove the build image on Docker host after pushing to the private Docker registry. Configure the Docker task (e.g., Remove Clients Build Image after Pushing) as follows:
+
+![](<media/removetestingclientsimage.png>)
+
+* **Action**: Select **Run a Docker command**.
+* **Command**: Enter the following command line, and replace the image-name with your image name (e.g., **myregistrydomain.com/clients:$(Build.BuildId)**):
+
+    ```
+    rmi <image-name>
+    ```
+
+**Step 18.** Save the build definition, and then click the **Queue new build** button.
+
+![](<media/saveandbuild.png>)
+
+Select **Default** as **Queue**, **master** as **Branch**, and then click **OK**.
+
+![](<media/queuebuildconfig.png>)
+
+**Step 19.** Once the build is done, click on the build step **Inspect Clients Container** to view the inspection results for **Clients** container.
+
+![](<media/inspectionresult.png>)
+
+**Step 20.** Click on the build step **Scan Security Vulnerabilities for Images and Containers** to view the scan results for images and containers.
+
+![](<media/scanresult.png>)
+
+**Step 21.** Open the Azure Blob Storage we created in previous task to store Docker images, you can see the new **clients** image.
+
+![](<media/clientsimage.png>)
+
+**Step 22.** Repeat the above steps for **Order** and **Database** components if you wish to practice more.
+
```

Comment:
Please add a sentence on the fact that we can optimize this workflow by creating a custom script when we call the docker inspect and docker bench command by rejecting the build and the push if something is wrong during the scan.